### DISCUSSION TITLE
Crash/Error using Biologically-inspired Dataset split

### DISCUSSION THREAD

**Discussion by CarrieASims** [cnr-isti-vclab/TagLab#109] [Q&A]
TagLab is crashing when I try to run the Biologically-Inspired dataset split when Exporting a New Training Dataset
This is the error:

PS E:\5_Taglab\TagLab-2022.11.3> py taglab.py
TagLab 2023.5.17
Traceback (most recent call last):
  File "E:\5_Taglab\TagLab-2022.11.3\taglab.py", line 4060, in exportNewDataset
    new_dataset.setupAreas(mode.upper(), target_classes)
  File "E:\5_Taglab\TagLab-2022.11.3\source\NewDataset.py", line 667, in setupAreas
    val_area, test_area = self.findAreas(target_classes=target_classes)
  File "E:\5_Taglab\TagLab-2022.11.3\source\NewDataset.py", line 397, in findAreas
    landscape_number, landscape_coverage, landscape_PSCV = self.calculateMetrics([0, 0, map_w, map_h], target_classes)
  File "E:\5_Taglab\TagLab-2022.11.3\source\NewDataset.py", line 291, in calculateMetrics
    if self.frequencies[key] > 0.0:
KeyError: (0, 'Background')
PS E:\5_Taglab\TagLab-2022.11.3>

I have limited understanding but seems it has an issue with the class Background, which is set default as Empty in the TagLab interface. 

Any assistance/enlightenment is most appreciated.
thanks

**Comment by maxcorsini** [comment]
Hi CarrieASims, 

This is a bug of some recent changes. I suggest that you export the dataset without using this option. If the problem remains, let us know.

**Reply by CarrieASims** [reply]
Grazie mille. I can extract as vertical or horizontal without issue for now. Look forward to the fix :)

**Reply by damarislanders** [reply]
Hi @maxcorsini,

I am getting the same issue. Not a problem as with @CarrieASims - I will use vertically split for now. Thought I would let you know.

Kind regards,

Damaris
